### PR TITLE
Sp spt idempotency

### PR DESCRIPTION
**Pull request by nabhajit-ray**
### Description
[Sp spt idempotency]

### Issues Resolved
[List any issues this PR will resolve. e.g., Fixes #01]

### Check List
- [ ] New functionality includes testing.
  - [ ]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/resource_server_profile_template.go [hunk 1]
```diff
@@ -1791,6 +1784,7 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
 		return sptError
 	}
 	return resourceServerProfileTemplateRead(d, meta)
+	//return nil
```
Comment (AsisBagga) [inline oneview/resource_server_profile_template.go:1787]: please remove these commented line.

File: oneview/resource_server_profile_template.go [hunk 2]
```diff
@@ -132,19 +135,16 @@ func resourceServerProfileTemplate() *schema.Resource {
 			},
 			"connection_settings": {
```
Comment (AsisBagga) [inline oneview/resource_server_profile_template.go:136]: please remove the changes in connection_settings...

**Review by AsisBagga — APPROVED** [review]
LGTM

File: oneview/resource_server_profile_template.go [hunk 3]
```diff
@@ -1216,13 +1211,15 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
...
-		MACType:            d.Get("mac_type").(string),
-		HideUnusedFlexNics: d.Get("hide_unused_flex_nics").(bool),
+		Name:                     d.Get("name").(string),
+		Type:                     d.Get("type").(string),
+		Affinity:                 d.Get("affinity").(string),
+		SerialNumberType:         d.Get("serial_number_type").(string),
+		WWNType:                  d.Get("wwn_type").(string),
+		MACType:                  d.Get("mac_type").(string),
+		HideUnusedFlexNics:       d.Get("hide_unused_flex_nics").(bool),
+		Description:              d.Get("description").(string),
```
Comment (AvijitOS) [inline oneview/resource_server_profile_template.go:1221]: i can see in schema description is set as only optional but server_profile_description set as both optional and computed any reason?

File: oneview/resource_server_profile_template.go [hunk 4]
```diff
@@ -1216,13 +1211,15 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
...
+		ServerProfileDescription: d.Get("server_profile_description").(string),
```
Comment (AvijitOS) [inline oneview/resource_server_profile_template.go:1222]: also i don't see scopes_uri related changes done(as u have done in EG PR), i think asis missed it

File: oneview/resource_server_profile_template.go [hunk 5]
```diff
@@ -1827,7 +1824,6 @@ func resourceServerProfileTemplateRead(d *schema.ResourceData, meta interface{})
 	}
 	d.Set("etag", spt.ETAG)
 	d.Set("hide_unused_flex_nics", spt.HideUnusedFlexNics)
-	d.Set("initial_scope_uris", spt.InitialScopeUris)
```
Comment (AvijitOS) [inline oneview/resource_server_profile_template.go:1830]: i can under volume also we have initial_scope_uris, not sure if we handle that as well. But there is one thing: if volume created as part of SP then initial_scope_uris should not be provided

File: oneview/resource_server_profile_template.go (hunk 3 above)
Comment (nabhajit-ray) [reply oneview/resource_server_profile_template.go:1221]: Fixed

File: oneview/resource_server_profile_template.go [hunk 6]
```diff
@@ -47,7 +50,7 @@ func resourceServerProfileTemplate() *schema.Resource {
 						},
 						"manage_bios": {
 							Type:     schema.TypeBool,
-							Required: true,
+							Optional: true,
```
[the remark this answers is not in the document]
Comment (nabhajit-ray) [reply oneview/resource_server_profile_template.go]: fixed

File: oneview/resource_server_profile_template.go (hunk 4 above)
Comment (nabhajit-ray) [reply oneview/resource_server_profile_template.go:1219]: fixed

File: oneview/resource_server_profile_template.go (hunk 5 above)
Comment (nabhajit-ray) [reply oneview/resource_server_profile_template.go:1830]: Not fixing it as it is part of san storage and we have not looked into it.

**Review by AvijitOS — APPROVED** [review]

**Pull request merged**